query/src/sql/parser/transformer/transform_sqlparser.rs

```diff
@@ -200,11 +200,24 @@ impl TransformerSqlparser {
         })
     }
 
-    fn transform_from(&self, orig_ast: &[TableWithJoins]) -> Result<Vec<TableReference>> {
-        orig_ast
+    fn transform_from(&self, orig_ast: &[TableWithJoins]) -> Result<TableReference> {
+        let mut table_refs: Vec<TableReference> = orig_ast
             .iter()
             .map(|v| self.transform_table_with_joins(v))
-            .collect()
+            .collect::<Result<_>>()?;
+        if table_refs.len() == 1 {
+            Ok(table_refs.drain(..).next().unwrap())
```

Comment:
looks good. But maybe we can reduce the use of `unwrap()`?

Reply:
I think so, do you have better idea?

Reply:
Not yet, let's merge first